src/classes/spot.ts

```diff
@@ -27,7 +28,7 @@ export class Spot {
         controller: AbortController,
     ): Promise<void> {
         try {
-            this.currentTask = crypto.randomUUID();
+            this.currentTask = uuidv4();
```

Comment:
Good catch, this might not work certain cases. However, it might be slower. I'm fine with switching